api/src/main/java/com/github/retrooper/packetevents/wrapper/play/client/WrapperPlayClientSetDifficulty.java

```diff
@@ -37,17 +37,17 @@ public WrapperPlayClientSetDifficulty(Difficulty difficulty) {
 
     @Override
     public void read() {
-        this.difficulty = Difficulty.getById(readByte());
+        this.difficulty = Difficulty.getById(readUnsignedByte());
     }
 
     @Override
-    public void copy(WrapperPlayClientSetDifficulty wrapper) {
-        this.difficulty = wrapper.difficulty;
+    public void write() {
+        writeByte(difficulty.getId());
```

Comment:
why is this writeByte and not writeUnsignedByte if the read method uses unsigned?

Reply:
```java
   public ClientboundChangeDifficultyPacket(FriendlyByteBuf var1) {
      this.difficulty = Difficulty.byId(var1.readUnsignedByte());
      this.locked = var1.readBoolean();
   }

   public void write(FriendlyByteBuf var1) {
      var1.writeByte(this.difficulty.getId());
      var1.writeBoolean(this.locked);
   }
```
Ask mojang please